uk.co.compucorp.civicrm.tasksassignments/js/test/services/task_test.js

```diff
@@ -0,0 +1,246 @@
+define([
+  'mocks/data/task',
+  'tasks-assignments/app'
+], function(TaskMock) {
+  'use strict';
+
+  describe('TaskService', function() {
+    var Task, TaskService, $provide, $httpBackend;
+
+    beforeEach(module('civitasks.appDashboard'));
+
+    beforeEach(inject(function(_Task_, _TaskService_, _$httpBackend_) {
+      Task = _Task_;
+      TaskService = _TaskService_;
+      $httpBackend = _$httpBackend_;
+      spyOn(Task, 'save').and.callThrough();
+    }));
+
+    beforeEach(function() {
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+      $httpBackend.whenGET(/action=get&entity=Task/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+    });
+
+    describe("save()", function() {
+      beforeEach(function() {
+        $httpBackend.whenPOST(/rest&action=get&entity=Task/).respond({
+          is_error: 0,
+          values: [TaskMock.task]
+        });
+        TaskService.save(TaskMock.task);
+        $httpBackend.flush();
+      });
+
+      it('makes call to save()', function() {
+        expect(Task.save).toHaveBeenCalled();
+      });
+
+      it('calls save() with correct data', function() {
+        var params = {
+          action: 'create',
+          entity: 'Task',
+          sequential: 1,
+          debug: true,
+          json: Object({
+            sequential: 1,
+            debug: true,
+            activity_date_time: '2017-04-21',
+            assignee_contact_id: ['6'],
+            source_contact_id: '205',
+            target_contact_id: ['10'],
+            details: '<p>Detailed description of the task</p>',
+            case_id: null,
+            activity_type_id: '88',
+            subject: 'Sample Task Subject'
```

Comment:
You are using mocked data
```js
$httpBackend.whenPOST(/rest&action=get&entity=Task/).respond({
  is_error: 0,
  values: [TaskMock.task]
});
```
so there is no need for you to hardcode these values. It means that if for some reason you are going to update the mocked data, this test will fail because the values won't match anymore.

Instead just reference that mocked data directly